src/search.c

```diff
@@ -379,6 +405,18 @@ static bool parse_options(int argc, char **argv)
 				scope = 'o';
 			}
 
+			break;
+		case 't':
+			if (!optarg) {
```

Comment:
getopt_long will return an error indication if ```-t``` doesn't have an argument

Reply:
We expect it to. `--top` requires `NUM` as an argument, or am I misunderstanding your comment?

Reply:
If you say ```wibble -t``` (i.e. without an argument to tthen getopt_long will return `?` rather than `t`. If it gets to the ```case 't':``` case then optarg will point somewhere.